File: backend/src/main/java/com/school/management/entity/MajorDirection.java
package com.school.management.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 专业方向实体类
 * 同一专业的不同培养层次
 */
@Data
@TableName("major_directions")
public class MajorDirection {

    /**
     * 主键ID
     */
    @TableId(value = "id", type = IdType.AUTO)
    private Long id;

    /**
     * 所属专业ID
     */
    private Long majorId;

    /**
     * 方向名称(如:3+2高级工班)
     */
    private String directionName;

    /**
     * 方向编码(如:CN_3P2_ADV)
     */
    private String directionCode;

    /**
     * 层次(中级工/高级工/预备技师/技师)
     */
    @TableField("level")
    private String level;

    /**
     * 学制(年)
     */
    private Integer years;

    /**
     * 是否分段注册(0=否,1=是)
     */
    private Integer isSegmented;

    /**
     * 第一阶段层次
     */
    private String phase1Level;

    /**
     * 第一阶段学制(年)
     */
    private Integer phase1Years;

    /**
     * 第二阶段层次
     */
    private String phase2Level;

    /**
     * 第二阶段学制(年)
     */
    private Integer phase2Years;

    /**
     * 备注
     */
    private String remarks;

    /**
     * 创建时间
     */
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    /**
     * 更新时间
     */
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;

    /**
     * 创建人ID
     */
    private Long createdBy;

    /**
     * 更新人ID
     */
    private Long updatedBy;

    /**
     * 删除标志(0=未删除,1=已删除)
     */
    @TableLogic
    private Integer deleted;

    /**
     * 专业名称(非数据库字段)
     */
    @TableField(exist = false)
    private String majorName;

    /**
     * 获取显示用的层次描述
     */
    public String getLevelDisplay() {
        if (isSegmented != null && isSegmented == 1) {
            return phase1Level + " → " + phase2Level;
        }
        return level;
    }

    /**
     * 获取显示用的学制描述
     */
    public String getYearsDisplay() {
        if (isSegmented != null && isSegmented == 1) {
            return phase1Years + "+" + phase2Years + "年";
        }
        return years + "年";
    }
}

File: frontend/src/api/majorDirection.ts
import { http } from '@/utils/request'

export interface MajorDirection {
  id?: number
  majorId: number
  directionName: string
  directionCode: string
  level: string           // 层次：中级工/高级工/预备技师/技师
  years: number           // 学制（年）
  isSegmented?: number    // 是否分段注册：0-否，1-是
  phase1Level?: string    // 第一阶段层次
  phase1Years?: number    // 第一阶段年数
  phase2Level?: string    // 第二阶段层次
  phase2Years?: number    // 第二阶段年数
  remarks?: string
  createdAt?: string
  updatedAt?: string
  createdBy?: number
  updatedBy?: number
  majorName?: string      // 关联显示
  // 显示用辅助字段
  levelDisplay?: string   // 层次显示（如：中级工 → 高级工）
  yearsDisplay?: string   // 学制显示（如：3+2年）
}

export interface MajorDirectionQueryParams {
  pageNum?: number
  pageSize?: number
  majorId?: number
  directionName?: string
  level?: string
}

// 获取专业方向列表(分页)
export const getMajorDirectionList = (params: MajorDirectionQueryParams) => {
  return http.get<{
    records: MajorDirection[]
    total: number
    pages: number
  }>('/major-directions', { params })
}

// 获取所有专业方向
export const getAllDirections = () => {
  return http.get<MajorDirection[]>('/major-directions/all')
}

// 根据专业ID获取专业方向列表
export const getDirectionsByMajor = (majorId: number | string) => {
  return http.get<MajorDirection[]>(`/major-directions/major/${majorId}`)
}

// 获取专业方向详情
export const getMajorDirectionDetail = (id: number | string) => {
  return http.get<MajorDirection>(`/major-directions/${id}`)
}

// 新增专业方向
export const addMajorDirection = (data: MajorDirection) => {
  return http.post<MajorDirection>('/major-directions', data)
}

// 更新专业方向
export const updateMajorDirection = (id: number | string, data: MajorDirection) => {
  return http.put(`/major-directions/${id}`, data)
}

// 删除专业方向
export const deleteMajorDirection = (id: number | string) => {
  return http.delete(`/major-directions/${id}`)
}

// 批量删除专业方向
export const batchDeleteDirections = (ids: (number | string)[]) => {
  return http.delete('/major-directions/batch', { data: ids })
}
